### PR TITLE
Submachinegun briefcase 

### PR DESCRIPTION
**Pull request by cogreaper**
## About the PR 
It implements a submachine gun hidden in a briefcase. It just adds a gun and a sprite for it. This doesn't seem to break any other interactions or items. As the sprite is in the storage items file it may impact those.

## Why's this needed? 
The idea was to have a traitor gun that is thematically more interesting than the very powerful items that are available to them. It is also an interesting semi stealthy weapon as it doesn't give itself away at first glance but is rather obvious when fired. I feel like this gun comes in as a hybrid between the stealth of the STL orion and the capability of the CPA predator while not being as good at either. Also it looks cool and would make for an interesting drop in an adventure zone. It currently is not obtainable outside of admin commands.

### CODE REVIEW

**Comment by pali6** [comment]
If it's not actually player accessible it doesn't need a changelog entry.

File: code/obj/item/gun/kinetic.dm [hunk 1]
```diff
@@ -1455,3 +1455,86 @@
...
+					return
+
+	canshoot()
+		if(open==1)
+			return 0
+		else
+			.=..()
+
+	update_icon()
+		if(open==1)
```
Comment (ZeWaka) [inline code/obj/item/gun/kinetic.dm:1537]: no need to check ==1, just check the var

Comment (pali6) [reply code/obj/item/gun/kinetic.dm:1537]:
```suggestion
		if(open)
```

File: code/obj/item/gun/kinetic.dm [hunk 2]
```diff
@@ -1455,3 +1455,86 @@
...
+			open=1
+			update_icon()
+			if ((src.loc == user) && user.find_in_hand(src)) // Make sure it's not on the belt or in a backpack.
+				src.add_fingerprint(user)
+				if (src.sanitycheck(0, 1) == 0)
+					user.show_text("You can't unload this gun.", "red")
+					return
+				if ((src.casings_to_eject > 0) && src.current_projectile.casing)
+					if (src.sanitycheck(1, 0) == 0)
+						logTheThing("debug", usr, null, "<b>Convair880</b>: [usr]'s gun ([src]) ran into the casings_to_eject cap, aborting.")
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1519]:
```suggestion
						logTheThing("debug", usr, null, "<b>Convair880</b>: [user]'s gun ([src]) ran into the casings_to_eject cap, aborting.")
```

File: code/obj/item/gun/kinetic.dm [hunk 3]
```diff
@@ -1455,3 +1455,86 @@
...
+					else
+						user.show_text("You eject [src.casings_to_eject] casings from [src].", "red")
+						src.ejectcasings()
+						return
+				else
+					user.show_text("[src] is empty!", "red")
+					return
+
+	canshoot()
+		if(open==1)
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1531]:
```suggestion
		if(open)
```

File: code/obj/item/gun/kinetic.dm [hunk 4]
```diff
@@ -1455,3 +1455,86 @@
...
+	attack_self(mob/user)
+		if(open==1)
+			open=0
+			update_icon()
+			boutput(user, "<span class='alert'>You close the [src]!</span>")
+		else
+			boutput(user,"<span class='alert'>You open the [src].</span>")
+			open=1
+			update_icon()
+			if ((src.loc == user) && user.find_in_hand(src)) // Make sure it's not on the belt or in a backpack.
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1512]:
```suggestion
			if (src.loc == user && user.find_in_hand(src)) // Make sure it's not on the belt or in a backpack.
```

File: code/obj/item/gun/kinetic.dm [hunk 5]
```diff
@@ -1455,3 +1455,86 @@
...
+			boutput(user, "<span class='alert'>You can't access the gun inside the [src] while it's closed! You'll have to open the [src]!</span>")
+
+	attack_self(mob/user)
+		if(open==1)
+			open=0
+			update_icon()
+			boutput(user, "<span class='alert'>You close the [src]!</span>")
+		else
+			boutput(user,"<span class='alert'>You open the [src].</span>")
+			open=1
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1510]:
```suggestion
			open = TRUE
```

File: code/obj/item/gun/kinetic.dm [hunk 6]
```diff
@@ -1455,3 +1455,86 @@
...
+
+	attackby(obj/item/ammo/bullets/b as obj, mob/user as mob)
+		if(open==1)
+			.=..()
+		else
+			boutput(user, "<span class='alert'>You can't access the gun inside the [src] while it's closed! You'll have to open the [src]!</span>")
+
+	attack_self(mob/user)
+		if(open==1)
+			open=0
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1505]:
```suggestion
			open = FALSE
```

File: code/obj/item/gun/kinetic.dm [hunk 7]
```diff
@@ -1455,3 +1455,86 @@
...
+			boutput(user,"<span class='alert'>You can't unload the [src] while it is closed.</span>")
+
+	attackby(obj/item/ammo/bullets/b as obj, mob/user as mob)
+		if(open==1)
+			.=..()
+		else
+			boutput(user, "<span class='alert'>You can't access the gun inside the [src] while it's closed! You'll have to open the [src]!</span>")
+
+	attack_self(mob/user)
+		if(open==1)
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1504]:
```suggestion
		if(open)
```

File: code/obj/item/gun/kinetic.dm [hunk 8]
```diff
@@ -1455,3 +1455,86 @@
...
+	attack_hand(mob/user as mob)
+		if(!user.find_in_hand(src))
+			..() //this works, dont touch it
+		else if(open==1)
+			.=..()
+		else
+			boutput(user,"<span class='alert'>You can't unload the [src] while it is closed.</span>")
+
+	attackby(obj/item/ammo/bullets/b as obj, mob/user as mob)
+		if(open==1)
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1498]:
```suggestion
		if(open)
```

File: code/obj/item/gun/kinetic.dm [hunk 9]
```diff
@@ -1455,3 +1455,86 @@
...
+
+	attack_hand(mob/user as mob)
+		if(!user.find_in_hand(src))
+			..() //this works, dont touch it
+		else if(open==1)
+			.=..()
+		else
+			boutput(user,"<span class='alert'>You can't unload the [src] while it is closed.</span>")
+
+	attackby(obj/item/ammo/bullets/b as obj, mob/user as mob)
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1497]:
```suggestion
	attackby(obj/item/ammo/bullets/b as obj, mob/user)
```

File: code/obj/item/gun/kinetic.dm [hunk 10]
```diff
@@ -1455,3 +1455,86 @@
...
+	New()
+		ammo = new/obj/item/ammo/bullets/nine_mm_NATO
+		current_projectile = new/datum/projectile/bullet/nine_mm_NATO/burst
+		open=0
+		..()
+
+	attack_hand(mob/user as mob)
+		if(!user.find_in_hand(src))
+			..() //this works, dont touch it
+		else if(open==1)
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1492]:
```suggestion
		else if(open)
```

File: code/obj/item/gun/kinetic.dm [hunk 11]
```diff
@@ -1455,3 +1455,86 @@
...
+			.=..()
+		else
+			boutput(user,"<span class='alert'>You can't unload the [src] while it is closed.</span>")
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1495]:
```suggestion
			boutput(user, "<span class='alert'>You can't unload the [src] while it is closed.</span>")
```

File: code/obj/item/gun/kinetic.dm [hunk 12]
```diff
@@ -1455,3 +1455,86 @@
...
+	spread_angle = 2
+	can_dual_wield = 0
+	var/cases_to_eject = 0
+	var/open
+
+
+	New()
+		ammo = new/obj/item/ammo/bullets/nine_mm_NATO
+		current_projectile = new/datum/projectile/bullet/nine_mm_NATO/burst
+		open=0
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1486]:
```suggestion
```

File: code/obj/item/gun/kinetic.dm [hunk 13]
```diff
@@ -1455,3 +1455,86 @@
...
+	auto_eject = 0
+
+	flags =  FPRINT | TABLEPASS | CONDUCT | USEDELAY | EXTRADELAY
+	object_flags = NO_ARM_ATTACH
+	c_flags = NOT_EQUIPPED_WHEN_WORN | EQUIPPED_WHILE_HELD
+
+	spread_angle = 2
+	can_dual_wield = 0
+	var/cases_to_eject = 0
+	var/open
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1480]:
```suggestion
	var/open = FALSE
```

File: code/obj/item/gun/kinetic.dm [hunk 14]
```diff
@@ -1455,3 +1455,86 @@
...
+						return
+				else
+					user.show_text("[src] is empty!", "red")
+					return
+
+	canshoot()
+		if(open==1)
+			return 0
+		else
+			.=..()
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1534]:
```suggestion
			. = ..()
```

File: code/obj/item/gun/kinetic.dm [hunk 15]
```diff
@@ -1455,3 +1455,86 @@
...
+		else
+			boutput(user,"<span class='alert'>You open the [src].</span>")
+			open=1
+			update_icon()
+			if ((src.loc == user) && user.find_in_hand(src)) // Make sure it's not on the belt or in a backpack.
+				src.add_fingerprint(user)
+				if (src.sanitycheck(0, 1) == 0)
+					user.show_text("You can't unload this gun.", "red")
+					return
+				if ((src.casings_to_eject > 0) && src.current_projectile.casing)
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1517]:
```suggestion
				if (src.casings_to_eject > 0 && src.current_projectile.casing)
```

File: code/obj/item/gun/kinetic.dm [hunk 16]
```diff
@@ -1455,3 +1455,86 @@
...
+					if (src.sanitycheck(1, 0) == 0)
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1518]:
```suggestion
					if (!src.sanitycheck(1, 0))
```

File: code/obj/item/gun/kinetic.dm [hunk 17]
```diff
@@ -1455,3 +1455,86 @@
...
+			open=0
+			update_icon()
+			boutput(user, "<span class='alert'>You close the [src]!</span>")
+		else
+			boutput(user,"<span class='alert'>You open the [src].</span>")
+			open=1
+			update_icon()
+			if ((src.loc == user) && user.find_in_hand(src)) // Make sure it's not on the belt or in a backpack.
+				src.add_fingerprint(user)
+				if (src.sanitycheck(0, 1) == 0)
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1514]:
```suggestion
				if (!src.sanitycheck(0, 1))
```

File: code/obj/item/gun/kinetic.dm [hunk 18]
```diff
@@ -1455,3 +1455,86 @@
...
+		else
+			boutput(user, "<span class='alert'>You can't access the gun inside the [src] while it's closed! You'll have to open the [src]!</span>")
+
+	attack_self(mob/user)
+		if(open==1)
+			open=0
+			update_icon()
+			boutput(user, "<span class='alert'>You close the [src]!</span>")
+		else
+			boutput(user,"<span class='alert'>You open the [src].</span>")
```
Comment (pali6) [inline code/obj/item/gun/kinetic.dm:1509]:
```suggestion
			boutput(user, "<span class='alert'>You open the [src].</span>")
```

**Pull request merged**